ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/util/RateLimiterTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.util;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThrows;
+
+import io.confluent.ksql.rest.util.ConcurrencyLimiter.Decrementer;
+import io.confluent.ksql.util.KsqlException;
+import java.util.Collections;
+import java.util.Map;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.apache.kafka.common.metrics.Metrics;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RateLimiterTest {
+
+  public static final String METRIC_NAMESPACE = "test";
+
+  @Test
+  public void shouldSucceedUnderLimit() {
+    final Metrics metrics = new Metrics();
+    final Map<String, String> tags = Collections.emptyMap();
+    // It doesn't look like the underlying guava rate limiter has a way to control time, so we're
+    // just going to have to hope that these tests reliably run in under a second.
```

Comment:
This is a bit of a bummer. I'm not sure how they test this library at Google, since the logic is time sensitive, but there's no way to control the time.

Regardless, a full second is an extremely long time to call "checkLimit" a couple of times, so I think we'll be safe.